tests/spec/int_series/test_arithmetic.py

```diff
@@ -1,43 +1,34 @@
-from databuilder.query_language import Dataset, IdSeries, IntSeries, build_patient_table
-
-from ..helpers import transpose
-
-p = build_patient_table(
-    "p",
-    {
-        "patient_id": IdSeries,
-        "i": IntSeries,
-    },
-)
-
-
-def test_arithmetic(in_memory_engine, subtests):
-    in_memory_engine.setup(
-        {
-            p: (
-                [1, 101],
-                [2, None],
-            ),
-        }
+from ..tables import p
+
+table_data = {
+    p: """
+          |  i1 |  i2
+        --+-----+-----
+        1 | 101 | 111
+        2 | 201 |
+        """,
+}
+
+
+def test_negate(spec_test):
+    spec_test(
+        table_data,
+        -p.i2,
+        {1: -111, 2: None},
     )
 
-    dataset = Dataset()
-    dataset.use_unrestricted_population()
-    dataset.negate = -p.i
-    dataset.add1 = p.i + 1
-    dataset.add2 = 1 + p.i
-    dataset.subtract1 = p.i - 1
-    dataset.subtract2 = 1 - p.i
 
-    results = transpose(in_memory_engine.extract(dataset))
-
-    with subtests.test("negate"):
-        assert results["negate"] == {1: -101, 2: None}
+def test_add(spec_test):
+    spec_test(
+        table_data,
+        p.i1 + p.i2,
+        {1: 212, 2: None},
```

Comment:
I really like the pattern of making the arithmetic explicit in the expected results (`101 + 111` here).

Reply:
Fixed.